packages/override-tools/src/test/Manifest.test.ts

```diff
@@ -155,10 +155,208 @@ test('BaseFileNotFound', async () => {
 
   const expectedError: ValidationError = {
     type: 'baseFileNotFound',
-    file: 'foo/bar.js',
+    file: 'foo\\bar.js',
   };
 
   const testManifest = new Manifest(ourManifestData, ovrRepo, reactRepo);
   const errors = await testManifest.validate();
   expect(errors).toEqual([expectedError]);
 });
+
+test('HasOverride', async () => {
+  const manifest = new Manifest(testManifestData, ovrRepo, reactRepo);
+  expect(manifest.hasOverride(overrideFiles[0].filename)).toBe(true);
+});
+
+test('DoesNotHaveOverride', async () => {
+  const manifest = new Manifest(testManifestData, ovrRepo, reactRepo);
+  expect(manifest.hasOverride('Never gonna give you up')).toBe(false);
+});
+
+test('RemoveOverride', async () => {
+  const manifest = new Manifest(testManifestData, ovrRepo, reactRepo);
+
+  expect(manifest.hasOverride(overrideFiles[0].filename)).toBe(true);
+  expect(manifest.removeOverride(overrideFiles[0].filename)).toBe(true);
+  expect(manifest.hasOverride(overrideFiles[0].filename)).toBe(false);
+});
+
+test('CannotRemoveOverride', async () => {
+  const manifest = new Manifest(testManifestData, ovrRepo, reactRepo);
+  expect(manifest.removeOverride('Never gonna let you down')).toBe(false);
+});
+
+test('addOverrideSimple', async () => {
+  const manifest = new Manifest(testManifestData, ovrRepo, reactRepo);
```

Comment:
>const manifest = new Manifest(testManifestData, ovrRepo, reactRepo); [](start = 2, length = 68)

Wouldn't this create a manifest that already has overrides for overrideFiles[0], etc.? If my understanding is correct, then I think this test is not testing add but is testing modification of overrides.

Reply:
Same question for the other add* tests.

---
In reply to: [387356252](https://github.com/microsoft/react-native-windows/pull/4219#discussion_r387356252) [](ancestors = 387356252)

Reply:
Goof catch. I need to pass in an empty object instead of testManifestData.

---
In reply to: [387357199](https://github.com/microsoft/react-native-windows/pull/4219#discussion_r387357199) [](ancestors = 387357199,387356252)

Reply:
Looks like I did this for the tests near the bottom but completely blanked on these. Will fix right now.

---
In reply to: [387358176](https://github.com/microsoft/react-native-windows/pull/4219#discussion_r387358176) [](ancestors = 387358176,387357199,387356252)

Reply:
I completely missed the fact that this PR had auto merge... So could you please send out a separate CR for this? 😅

---
In reply to: [387359393](https://github.com/microsoft/react-native-windows/pull/4219#discussion_r387359393) [](ancestors = 387359393,387358176,387357199,387356252)